hazelcast/src/main/java/com/hazelcast/ringbuffer/impl/RingbufferContainer.java

```diff
@@ -464,8 +520,11 @@ public void writeData(ObjectDataOutput out) throws IOException {
     @Override
     @SuppressWarnings("unchecked")
     public void readData(ObjectDataInput in) throws IOException {
-        name = in.readUTF();
-        emptyRingWaitNotifyKey = new RingbufferWaitNotifyKey(name, EMPTY);
+        assert !in.getVersion().isUnknown();
```

Comment:
The `ReplicationOperation` which serialises/deserialises the container is `Versioned`. I could make the container `Versioned` as well.